llvm/lib/CheerpUtils/IdenticalCodeFolding.cpp

```diff
@@ -844,6 +849,29 @@ bool IdenticalCodeFolding::isStaticIndirectFunction(const llvm::Value* A)
 	return ce && ce->isCast() && isa<Function>(ce->getOperand(0));
 }
 
+bool IdenticalCodeFolding::isFunctionExternal(const llvm::Function* F)
+{
+	if (externalFunctionMapping.count(F))
+		return externalFunctionMapping[F];
+
+	bool functionExternal = false;
+	for (const Use& U : F->uses())
+	{
+		const User* user = U.getUser();
+		if (const Instruction* ci = dyn_cast<Instruction>(user))
+		{
+			if (ci->getFunction()->getSection() != StringRef("asmjs"))
+			{
+				externalFunctionMapping[F] = true;
```

Comment:
Prefer `emplace(F, true)` to `[F] = true`

Reply:
Makes sense, updated.